lightning_pose/data/datasets.py

```diff
@@ -330,12 +332,191 @@ def __getitem__(self, idx: int) -> HeatmapLabeledExampleDict:
+        self.do_context = do_context
+
+        csv_data={}
+        self.image_names = {}
+        self.keypoints = {}        
+        csv_row_sizes =  []
+        self.keypoint_names = []
+        self.keypoint_names_view = []
+        for csv_path in self.csv_paths:
+            if os.path.isfile(csv_path):
+                csv_file = csv_path
+            else:
+                csv_file = os.path.join(root_directory, csv_path)
+            if not os.path.exists(csv_file):
+                # step 2: assume csv_path is absolute
+                csv_file = csv_path
+                if not os.path.exists(csv_file):
+                    # step 3: assume dlc directory structure
+                    import glob
+
+                    glob_path = os.path.join(
+                        root_directory,
+                        "training-data",
+                        "iteration-0",
+                        "*",  # wildcard handles proj-specific dlc naming conventions
+                        csv_path,
+                    )
+                    options = glob.glob(glob_path)
+                    if not options or not os.path.exists(options[0]):
+                        raise FileNotFoundError("Could not find csv file!")
+                    csv_file = options[0]
+            view = csv_path.split("/")[-1].split(".")[0]
+            csv_data[view] = pd.read_csv(csv_file, header=header_rows, index_col=0)
+            self.keypoint_names.extend(get_keypoint_names(csv_file=csv_file, header_rows=header_rows))
+            # self.keypoint_names_view.extend([view]*len()) TODO: book keeping
+            self.image_names[view] = list(csv_data[view].index)
+            csv_row_sizes.append(len(self.image_names[view]))
+
+            self.keypoints[view] = csv_data[view].to_numpy().astype(np.float32)
+            # self.keypoints[view] = csv_data[view].to_numpy()
+            # convert to x,y coordinates
+                    
+        self.keypoints = torch.tensor(np.concatenate(list(self.keypoints.values()), axis=1), dtype=torch.float32)
+        self.keypoints = self.keypoints.reshape(self.keypoints.shape[0], -1, 2)
+        self.data_lenght = len(self.image_names[view])
+        
+        if len(set(csv_row_sizes)) != 1:
+            raise LookupError("number of rows in the CSV files do not match!")        
+
+        # send image to tensor and normalize
+        pytorch_transform_list = [
+            transforms.ToTensor(),
+            transforms.Normalize(mean=_IMAGENET_MEAN, std=_IMAGENET_STD),
+        ]
+        self.pytorch_transform = transforms.Compose(pytorch_transform_list)
+
+        # # keypoints has been already transformed above
+
+        # Compute heatmaps as preprocessing step
+        self.num_targets = torch.numel(self.keypoints[0])
+        self.num_keypoints = self.num_targets // 2
+        self.label_heatmaps = None  # populated by `self.compute_heatmaps()`
+        self.compute_heatmaps()
+
```

Comment:
add height/width property; assume it is the same for all views, so we just need to access that property from a single view

```
@property
def height(self) -> int:
    return self.datasets[self.view_names[0]].height
```

Reply:
```
@property
def num_views(self) -> int:
    return len(self.datasets)
```

Reply:
```
def __len__():
    return len(self.datasets[self.view_names[0]])
```